File: Winforms Example 9 - Table Properties Dialog/frmTableProperties.vb
Imports System.Text

Public Class frmTableProperties

    Private m_Adding As Boolean
    Public Cancelled As Boolean = True

    Public Property Adding As Boolean
        Get
            Return m_Adding
        End Get
        Set(value As Boolean)
            Me.txtColumns.Enabled = value
            Me.txtRows.Enabled = value
            m_Adding = value
        End Set
    End Property

    Private Sub btnOK_Click(sender As Object, e As EventArgs) Handles btnOK.Click
        Cancelled = False
        Me.Hide()
    End Sub

    Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        Cancelled = True
        Me.Hide()
    End Sub

    Private Sub btnBorderColor_Click(sender As Object, e As EventArgs) Handles btnBorderColor.Click

        Dim oColor As Color = GetColor()

        If Not oColor = Nothing Then

            Dim sb As StringBuilder = New StringBuilder()
            sb.AppendFormat("#")
            sb.Append(BitConverter.ToString(New Byte() {oColor.R}))
            sb.Append(BitConverter.ToString(New Byte() {oColor.G}))
            sb.Append(BitConverter.ToString(New Byte() {oColor.B}))

            txtBorderColor.Text = sb.ToString
        End If

    End Sub

    Private Sub btnBackgroundColor_Click(sender As Object, e As EventArgs) Handles btnBackgroundColor.Click

        Dim oColor As Color = GetColor()

        If Not oColor = Nothing Then

            Dim sb As StringBuilder = New StringBuilder()
            sb.AppendFormat("#")
            sb.Append(BitConverter.ToString(New Byte() {oColor.R}))
            sb.Append(BitConverter.ToString(New Byte() {oColor.G}))
            sb.Append(BitConverter.ToString(New Byte() {oColor.B}))

            txtBackgroundColor.Text = sb.ToString
        End If

    End Sub

    Private Function GetColor() As Color

        Dim oColorDialog As New ColorDialog
        Dim oDialogResult As DialogResult

        oDialogResult = oColorDialog.ShowDialog(Me)

        If oDialogResult = DialogResult.OK Then
            Return oColorDialog.Color
        Else
            Return Nothing
        End If

    End Function

    Private Sub chkApplyCellProperties_CheckedChanged(sender As Object, e As EventArgs) Handles chkApplyCellProperties.CheckedChanged

        grpCellProperties.Enabled = Me.chkApplyCellProperties.Checked

    End Sub

    Private Sub btnCellBorderColor_Click(sender As Object, e As EventArgs) Handles btnCellBorderColor.Click

        Dim oColor As Color = GetColor()

        If Not oColor = Nothing Then

            Dim sb As StringBuilder = New StringBuilder()
            sb.AppendFormat("#")
            sb.Append(BitConverter.ToString(New Byte() {oColor.R}))
            sb.Append(BitConverter.ToString(New Byte() {oColor.G}))
            sb.Append(BitConverter.ToString(New Byte() {oColor.B}))

            txtCellBorderColor.Text = sb.ToString
        End If

    End Sub

    Private Sub btnCellBackgroundColor_Click(sender As Object, e As EventArgs) Handles btnCellBackgroundColor.Click

        Dim oColor As Color = GetColor()

        If Not oColor = Nothing Then

            Dim sb As StringBuilder = New StringBuilder()
            sb.AppendFormat("#")
            sb.Append(BitConverter.ToString(New Byte() {oColor.R}))
            sb.Append(BitConverter.ToString(New Byte() {oColor.G}))
            sb.Append(BitConverter.ToString(New Byte() {oColor.B}))

            txtCellBackgroundColor.Text = sb.ToString
        End If

    End Sub
End Class

File: Winforms Example 09 - Table Properties Dialog/frmTableProperties.cs
using System;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace Winforms_Example_09___Table_Properties_Dialog_CS
{
    public partial class frmTableProperties : Form
    {

        private bool m_Adding;
        public bool Cancelled = true;

        public frmTableProperties()
        {
            InitializeComponent();
        }

        public bool Adding
        {
            get
            {
                return m_Adding;
            }
            set
            {
                this.txtColumns.Enabled = value;
                this.txtRows.Enabled = value;
                m_Adding = value;
            }
        }

        private void btnOK_Click(object sender, EventArgs e)
        {
            Cancelled = false;
            this.Hide();
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            Cancelled = true;
            this.Hide();
        }

        private void btnBorderColor_Click(object sender, EventArgs e)
        {
            Color oColor = GetColor();

            if (oColor != null/* TODO Change to default(_) if this is not a reference type */ )
            {
                StringBuilder sb = new StringBuilder();
                sb.AppendFormat("#");
                sb.Append(BitConverter.ToString(new byte[] { oColor.R }));
                sb.Append(BitConverter.ToString(new byte[] { oColor.G }));
                sb.Append(BitConverter.ToString(new byte[] { oColor.B }));

                txtBorderColor.Text = sb.ToString();
            }
        }

        private void btnBackgroundColor_Click(object sender, EventArgs e)
        {
            Color oColor = GetColor();

            if (oColor != null/* TODO Change to default(_) if this is not a reference type */ )
            {
                StringBuilder sb = new StringBuilder();
                sb.AppendFormat("#");
                sb.Append(BitConverter.ToString(new byte[] { oColor.R }));
                sb.Append(BitConverter.ToString(new byte[] { oColor.G }));
                sb.Append(BitConverter.ToString(new byte[] { oColor.B }));

                txtBackgroundColor.Text = sb.ToString();
            }
        }

        private Color GetColor()
        {
            ColorDialog oColorDialog = new ColorDialog();
            DialogResult oDialogResult;

            oDialogResult = oColorDialog.ShowDialog(this);

            if (oDialogResult == DialogResult.OK)
                return oColorDialog.Color;
            else
                return Color.Empty/* TODO Change to default(_) if this is not a reference type */;
        }

        private void chkApplyCellProperties_CheckedChanged(object sender, EventArgs e)
        {
            grpCellProperties.Enabled = this.chkApplyCellProperties.Checked;
        }

        private void btnCellBorderColor_Click(object sender, EventArgs e)
        {
            Color oColor = GetColor();

            if (oColor != null/* TODO Change to default(_) if this is not a reference type */ )
            {
                StringBuilder sb = new StringBuilder();
                sb.AppendFormat("#");
                sb.Append(BitConverter.ToString(new byte[] { oColor.R }));
                sb.Append(BitConverter.ToString(new byte[] { oColor.G }));
                sb.Append(BitConverter.ToString(new byte[] { oColor.B }));

                txtCellBorderColor.Text = sb.ToString();
            }
        }

        private void btnCellBackgroundColor_Click(object sender, EventArgs e)
        {
            Color oColor = GetColor();

            if (oColor != null/* TODO Change to default(_) if this is not a reference type */ )
            {
                StringBuilder sb = new StringBuilder();
                sb.AppendFormat("#");
                sb.Append(BitConverter.ToString(new byte[] { oColor.R }));
                sb.Append(BitConverter.ToString(new byte[] { oColor.G }));
                sb.Append(BitConverter.ToString(new byte[] { oColor.B }));

                txtCellBackgroundColor.Text = sb.ToString();
            }
        }

        private void cboBorderWidth_SelectedIndexChanged(object sender, EventArgs e)
        {

        }
    }
}
